datacleaner/datacleaner.py

```diff
@@ -76,7 +76,13 @@ def autoclean(input_dataframe, drop_nans=False, copy=False, encoder=None,
         try:
             input_dataframe[column].fillna(input_dataframe[column].median(), inplace=True)
         except TypeError:
-            input_dataframe[column].fillna(input_dataframe[column].mode()[0], inplace=True)
+            most_frequent = input_dataframe[column].mode()
+            if len(most_frequent)>0:
+                input_dataframe[column].fillna(input_dataframe[column].mode()[0], inplace=True)
+            else:
+                input_dataframe[column].fillna(method='bfill', inplace=True)
+                input_dataframe[column].fillna(method='ffill', inplace=True)
```

Comment:
Hi I write like this because some scenario like below, when the first or the last line has nan value, only the ffill or bfill may be not enough:
```python
>>> import pandas as pd
>>> import numpy as np
>>> df=pd.DataFrame({"a":[np.nan,1,1],"b":[2,2,np.nan]})
>>> df
     a    b
0  NaN  2.0
1  1.0  2.0
2  1.0  NaN
>>> df["a"].fillna(method="ffill")
0    NaN
1    1.0
2    1.0
Name: a, dtype: float64
>>> df["b"].fillna(method="bfill")
0    2.0
1    2.0
2    NaN
Name: b, dtype: float64
>>> 
```